### ISSUE TITLE
unexpected zoom/scroll behavior in gallery with touch screens / touch pads

### ISSUE THREAD

**Issue by r10s** [deltachat/deltachat-desktop#2632]
in the gallery, images can be zoom meanwhile, which is great.

however, the used gestures are not standard and unexpected:

- the pinch/punch gesture works as expected and zoom the image
- once zoomed, however, a two finger slide does not move the image as usual but zooms further in and out or jitters around, depending on the direction 

we should either disable two finger sliding at all (i would not like that), or, much better, move the zoomed image around so that the user can actually enlarger every detail of an image and not only the center. the latter is also what eg. telegram is doing and most other apps i know about.

**Comment by Simon-Laux** [comment]
this might be difficult because there are multiple targets:
- mouse (scroll to zoom; drag to move)
- fancy gesture touchpad (pinch to zoom, two finger drag to move)
- standard touchpad without gestures (what's expected here? same as with mouse?)

**Comment by r10s** [comment]
yip, mouse works as expected, at a first glance (i do not have a mouse, but  click-move on the touchpad moves the image around).

maybe we can at least disable the two-finger thingie which is just wrong?

**Comment by Simon-Laux** [comment]
https://prc5.github.io/react-zoom-pan-pinch/?path=/story/examples-big-image--big-image

`wheel.wheelDisabled` fixes the strange behaviour on fancy apple touchpads, but then scrolling to zoom with a mouse does not work anymore.
`wheel.touchPadDisabled` could be what we need in theory but does not work on their demo site :(.

**Comment by r10s** [comment]
> I think the expected behaviour for the Mac touchpad would be:
>
>    2 finger diagonal pinching (moving together) for zooming
>    2 finger panning (moving together in one direction) for moving around in the zoomed in image

yes. but that is not limited to mac, there is nothing fancy in that any longer :) windows uses the same gestures (pinch/punch to zoom and two-fingers to scroll) and iirc i also used the same in many ubuntu apps.

**Comment by r10s** [comment]
to sum up: to fix the bug:

- disable zooming/jittering when doing two-finger-panning
- scrollwheel-zooming could be done by adding a qualifier - maybe `wheel.activationKeys` solves both points.

that would still not allow two-finger-panning, but also would not result in unexpected jittering when trying so.

panning can still be done by dragging.

of course, having real two-finger-panning would be nice as well, but that is beyond the bug fix.

**Comment by Jikstra** [comment]
Sorry but I don't really see this as an issue with any priority. If anyone feels like working on this that's of course appreciated but besides this it's an annoyance that only sooome people will experience. I think besides @treefits mac no other dev device even supports multi gestures on the hardware side.

**Comment by r10s** [comment]
i agree that this is not the most important bug. and yes, it is hard to target for devs not using touchpad supporting gestures.

but it is still pretty annoying and unusual ui/ux - and not only for sooome mac, but for most ppl using a touchpad - gestures are quite usual on all systems nowadays :)

as you probably cannot really see the issue then, here is a little video showing how things look like when zooming to a point (works) and then try to pan around (works not - results in zooming and jittering):

https://user-images.githubusercontent.com/9800740/159467122-d8b14765-5b68-4599-a428-409aa9e2b92e.mov

so, i think, that should be targeted somehow, see details above - but we can do that later and we can remove it from the 1.30 project.

**Comment by Simon-Laux** [comment]
https://github.com/prc5/react-zoom-pan-pinch/issues/370

**Comment by maxphilippov** [comment]
@r10s I don't use gestures a lot, but does Profile Image Selector work the way you describe? Could it be worth to just factor out zoom/pan code from there and apply it to Gallery?

**Comment by Simon-Laux** [comment]
> Profile Image Selector

On that note, we should check if the todo items in https://github.com/deltachat/deltachat-desktop/issues/3893 are still to do.

**Comment by r10s** [comment]
> I don't use gestures a lot, but does Profile Image Selector work the way you describe?

in the profile image selector, the zoom gestures are differently, but not correct.

in the jprofile image selector the zoom gestures are exactly swapped - the gesture that should make the image smaller (pinch in) makes it actually bigger:

https://github.com/user-attachments/assets/eb56d22a-ea03-49f2-a2ee-be028238bef6

this is totally unintuitive, imagine you have a sponge between the fingers, that would also get smaller :) and again, this is not my personal preference, but standard

the swapped behaviour also led to the idea that zooming is entirely impossible, btw, which led to adding the +/- buttons
